File: MainApp/src/main/java/com/sleticalboy/dailywork/encryption/SM4Context.java
package com.sleticalboy.dailywork.encryption;

public class SM4Context {

    /**
     * 加密模式
     */
    private int mode;

    private long[] sk;

    private boolean isPadding;

    public SM4Context() {
        this.mode = 1;
        this.isPadding = true;
        this.sk = new long[32];
    }

    public int getMode() {
        return mode;
    }

    public void setMode(int mode) {
        this.mode = mode;
    }

    public long[] getSk() {
        return sk;
    }

    public void setSk(long[] sk) {
        this.sk = sk;
    }

    public boolean isPadding() {
        return isPadding;
    }

    public void setPadding(boolean padding) {
        isPadding = padding;
    }

}

File: MainApp/src/main/java/com/sleticalboy/learning/encryption/SM4Context.kt
package com.sleticalboy.learning.encryption

class SM4Context {

  /**
   * 加密模式
   */
  private var mode = 1
  private var sk: LongArray
  private var isPadding = true

  fun getMode(): Int = mode

  fun setMode(mode: Int) {
    this.mode = mode
  }

  fun getSk(): LongArray = sk

  fun setSk(sk: LongArray) {
    this.sk = sk
  }

  fun isPadding(): Boolean = isPadding

  fun setPadding(padding: Boolean) {
    isPadding = padding
  }

  init {
    sk = LongArray(32)
  }
}